File: pset6/caesar/caesar.py
from cs50 import get_string
import sys

if len(sys.argv) != 2:
    print("Usage: ./caesar.py k")
else:
    print("plaintext: ", end='')
    p = get_string()
    print("ciphertext: ", end='')
k = int(sys.argv[1])
leng = len(p)
for i in range(leng):
    if i != leng - 1:
        if p[i].isalpha() == True and p[i].isupper() == True:
            print("{}".format(chr(((ord(p[i]) - 65 + k) % 26) + 65)), end='')
        elif p[i].isalpha() == True and p[i].isupper() == False:
            print("{}".format(chr(((ord(p[i]) - 97 + k) % 26) + 97)), end='')
        else:
            print("{}".format(p[i]), end='')
    else:
        if p[i].isalpha() == True and p[i].isupper() == True:
            print("{}".format(chr(((ord(p[i]) - 65 + k) % 26) + 65)))
        elif p[i].isalpha() == True and p[i].isupper() == False:
            print("{}".format(chr(((ord(p[i]) - 97 + k) % 26) + 97)))
        else:
            print("{}".format(p[i]))
    

File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main (int argc, string argv[])
{
    string p; /*declaring variable p*/
    if (argc != 2) /*checking if number of command-line argument is not equal to two*/
    {
    printf ("Usage: ./caesar k\n"); /*generating error message*/
    return 1;
    }
    else
        {
        printf ("plaintext: ");
        p = get_string();
        printf ("ciphertext: ");
        }
    int k = atoi (argv[1]); /*converting string to integer*/
    for (int i = 0, n = strlen (p); i < n; i++) /*checking each char of string*/
    {
            if ((isalpha(p[i])) && (isupper(p[i])))  /*checking if char is uppercase*/
            {
            printf ("%c", ((p[i] - 65 + k) % 26) + 65); /*shift uppercase chars*/
            }
            else if ((isalpha(p[i])) && (islower(p[i]))) /*checking if char is lowercase*/
            {
            printf ("%c", ((p[i] - 97 + k) % 26) + 97); /*shift lowercase chars*/   
            }
            else
            {
            printf ("%c", p[i]); /*don't shift chars*/
            }
    }
    printf ("\n");
}